File: aat-android/src/main/java/ch/bailu/aat/description/SlopeDescription.java
package ch.bailu.aat.description;

import android.content.Context;

import ch.bailu.aat_lib.description.ContentDescription;
import ch.bailu.aat_lib.gpx.GpxInformation;
import ch.bailu.aat_lib.gpx.attributes.AltitudeDelta;
import ch.bailu.aat_lib.resources.Res;

public class SlopeDescription extends ContentDescription {
    private String slope="0";

    public SlopeDescription(Context c) {
    }

    @Override
    public void onContentUpdated(int iid, GpxInformation info) {
        slope = info.getAttributes().get(AltitudeDelta.INDEX_SLOPE);
    }

    @Override
    public String getValue() {
        return slope;
    }

    public String getUnit() {
        return "%";
    }

    @Override
    public String getLabel() {
        return Res.str().d_slope();
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/description/SlopeDescription.kt
package ch.bailu.aat_lib.description

import ch.bailu.aat_lib.gpx.GpxInformation
import ch.bailu.aat_lib.gpx.attributes.AltitudeDelta
import ch.bailu.aat_lib.resources.Res

class SlopeDescription : ContentDescription() {
    private var slope = "0"
    override fun onContentUpdated(iid: Int, info: GpxInformation) {
        slope = info.getAttributes()[AltitudeDelta.INDEX_SLOPE]
    }

    override fun getValue(): String {
        return slope
    }

    override fun getUnit(): String {
        return "%"
    }

    override fun getLabel(): String {
        return Res.str().d_slope()
    }
}
